File: apps/android/app/src/main/java/com/rsr41/oracle/domain/model/HistoryRecord.kt
package com.rsr41.oracle.domain.model

/**
 * 히스토리 기록 모델
 * - payload에 JSON 형태로 상세 데이터를 저장하여 나중에 복원 가능하게 함
 */
data class HistoryRecord(
    val id: String,
    val type: HistoryType,
    val title: String,
    val summary: String, // resultSummary
    val payload: String, // resultDetail (JSON), kept name for compatibility but used as detail
    val inputSnapshot: String = "{}", // New field for input JSON
    val profileId: String?,
    val partnerProfileId: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val expiresAt: Long? = null, // New field for TTL
    val locale: String = "ko"
)

File: apps/ios/OracleIOS_Sources/Domain/Entities/HistoryRecord.swift
import Foundation

/// 히스토리 기록 모델
/// Android: domain/model/HistoryRecord.kt (lines 7-19)
struct HistoryRecord: Identifiable, Equatable, Codable {
    let id: String
    let type: HistoryType
    let title: String
    let summary: String
    let payload: String  // 상세 데이터 JSON
    let inputSnapshot: String  // 입력 정보 JSON
    let profileId: String?
    let partnerProfileId: String?
    let createdAt: Date
    let expiresAt: Date?
    let locale: String
    
    init(
        id: String = UUID().uuidString,
        type: HistoryType,
        title: String,
        summary: String,
        payload: String,
        inputSnapshot: String = "{}",
        profileId: String? = nil,
        partnerProfileId: String? = nil,
        createdAt: Date = Date(),
        expiresAt: Date? = nil,
        locale: String = "ko"
    ) {
        self.id = id
        self.type = type
        self.title = title
        self.summary = summary
        self.payload = payload
        self.inputSnapshot = inputSnapshot
        self.profileId = profileId
        self.partnerProfileId = partnerProfileId
        self.createdAt = createdAt
        self.expiresAt = expiresAt
        self.locale = locale
    }
}

/// 레거시 히스토리 아이템 (호환용)
/// Android: domain/model/HistoryItem.kt (lines 3-7)
struct HistoryItem: Identifiable, Equatable, Codable {
    let id: String
    let birthInfo: BirthInfo
    let result: SajuResult
}
